<a href="https://colab.research.google.com/github/Yenuli0808/CM2606-DATA_Engineering_CW/blob/main/NoteBooks/01_data_profiling.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Data Profiling**

---

The dataset was profiled to understand its schema, data types, missing values and duplication levels prior to pipeline design.

In [1]:
import pandas as pd
import numpy as np

# Set display options for better viewing
pd.set_option('display.max_columns', None)
pd.set_option('display.width', 1000)

In [4]:
# Load the dataset from GitHub
url="https://raw.githubusercontent.com/Yenuli0808/CM2606-DATA_Engineering_CW/main/Dataset/superstore.csv"

# Load the data
df = pd.read_csv(url)
print("✅ Dataset loaded successfully!")
print("\n=== Data set overview ===")
print(f"Dataset shape: {df.shape}")

✅ Dataset loaded successfully!

=== Data set overview ===
Dataset shape: (51290, 27)


In [5]:
# First look at the data
print("==== FIRST 5 ROWS ====\n")
display(df.head())

==== FIRST 5 ROWS ====



,Category,City,Country,Customer.ID,Customer.Name,Discount,Market,记录数,Order.Date,Order.ID,Order.Priority,Product.ID,Product.Name,Profit,Quantity,Region,Row.ID,Sales,Segment,Ship.Date,Ship.Mode,Shipping.Cost,State,Sub.Category,Year,Market2,weeknum
0,Office Supplies,Los Angeles,United States,LS-172304,Lycoris Saunders,0.0,US,1,2011-01-07 00:00:00.000,CA-2011-130813,High,OFF-PA-10002005,Xerox 225,9.3312,3,West,36624,19,Consumer,2011-01-09 00:00:00.000,Second Class,4.37,California,Paper,2011,North America,2
1,Office Supplies,Los Angeles,United States,MV-174854,Mark Van Huff,0.0,US,1,2011-01-21 00:00:00.000,CA-2011-148614,Medium,OFF-PA-10002893,"Wirebound Service Call Books, 5 1/2"" x 4""",9.2928,2,West,37033,19,Consumer,2011-01-26 00:00:00.000,Standard Class,0.94,California,Paper,2011,North America,4
2,Office Supplies,Los Angeles,United States,CS-121304,Chad Sievert,0.0,US,1,2011-08-05 00:00:00.000,CA-2011-118962,Medium,OFF-PA-10000659,"Adams Phone Message Book, Professional, 400 Me...",9.8418,3,West,31468,21,Consumer,2011-08-09 00:00:00.000,Standard Class,1.81,California,Paper,2011,North America,32
3,Office Supplies,Los Angeles,United States,CS-121304,Chad Sievert,0.0,US,1,2011-08-05 00:00:00.000,CA-2011-118962,Medium,OFF-PA-10001144,Xerox 1913,53.2608,2,West,31469,111,Consumer,2011-08-09 00:00:00.000,Standard Class,4.59,California,Paper,2011,North America,32
4,Office Supplies,Los Angeles,United States,AP-109154,Arthur Prichep,0.0,US,1,2011-09-29 00:00:00.000,CA-2011-146969,High,OFF-PA-10002105,Xerox 223,3.1104,1,West,32440,6,Consumer,2011-10-03 00:00:00.000,Standard Class,1.32,California,Paper,2011,North America,40


In [6]:
print("\n==== Dataset Info ====\n")
df.info()


==== Dataset Info ====

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51290 entries, 0 to 51289
Data columns (total 27 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Category        51290 non-null  object 
 1   City            51290 non-null  object 
 2   Country         51290 non-null  object 
 3   Customer.ID     51290 non-null  object 
 4   Customer.Name   51290 non-null  object 
 5   Discount        51290 non-null  float64
 6   Market          51290 non-null  object 
 7   记录数             51290 non-null  int64  
 8   Order.Date      51290 non-null  object 
 9   Order.ID        51290 non-null  object 
 10  Order.Priority  51290 non-null  object 
 11  Product.ID      51290 non-null  object 
 12  Product.Name    51290 non-null  object 
 13  Profit          51290 non-null  float64
 14  Quantity        51290 non-null  int64  
 15  Region          51290 non-null  object 
 16  Row.ID          51290 non-null  int64  
 17  Sales 

In [8]:
# Check column names
print("Columns in dataset:")
print(df.columns)

Columns in dataset:
Index(['Category', 'City', 'Country', 'Customer.ID', 'Customer.Name', 'Discount', 'Market', '记录数', 'Order.Date', 'Order.ID', 'Order.Priority', 'Product.ID', 'Product.Name', 'Profit', 'Quantity', 'Region', 'Row.ID', 'Sales', 'Segment', 'Ship.Date', 'Ship.Mode', 'Shipping.Cost', 'State', 'Sub.Category', 'Year', 'Market2', 'weeknum'], dtype='object')


In [10]:
# check missing values
print("\nMissing values per column:")
print(df.isnull().sum())


Missing values per column:
Category          0
City              0
Country           0
Customer.ID       0
Customer.Name     0
Discount          0
Market            0
记录数               0
Order.Date        0
Order.ID          0
Order.Priority    0
Product.ID        0
Product.Name      0
Profit            0
Quantity          0
Region            0
Row.ID            0
Sales             0
Segment           0
Ship.Date         0
Ship.Mode         0
Shipping.Cost     0
State             0
Sub.Category      0
Year              0
Market2           0
weeknum           0
dtype: int64


In [9]:
# Check missing values
missing_values = df.isnull().sum()

print("Missing values per column:")
print(missing_values[missing_values > 0])

Missing values per column:
Series([], dtype: int64)


In [11]:
# Check duplicate rows
duplicate_count = df.duplicated().sum()

print("Number of duplicate rows:", duplicate_count)

Number of duplicate rows: 0


In [7]:
print("\n===== SUMMARY STATS =====")
df.describe(include="all")


===== SUMMARY STATS =====


,Category,City,Country,Customer.ID,Customer.Name,Discount,Market,记录数,Order.Date,Order.ID,Order.Priority,Product.ID,Product.Name,Profit,Quantity,Region,Row.ID,Sales,Segment,Ship.Date,Ship.Mode,Shipping.Cost,State,Sub.Category,Year,Market2,weeknum
count,51290,51290,51290,51290,51290,51290.000000,51290,51290.0,51290,51290,51290,51290,51290,51290.000000,51290.000000,51290,51290.00000,51290.000000,51290,51290,51290,51290.000000,51290,51290,51290.000000,51290,51290.000000
unique,3,3636,147,4873,795,NaN,7,NaN,1430,25035,4,10292,3788,NaN,NaN,13,NaN,NaN,3,1464,4,NaN,1094,17,NaN,6,NaN
top,Office Supplies,New York City,United States,JG-158051,Muhammed Yedwab,NaN,APAC,NaN,2014-06-18 00:00:00.000,CA-2014-100111,Medium,OFF-AR-10003651,Staples,NaN,NaN,Central,NaN,NaN,Consumer,2014-11-22 00:00:00.000,Standard Class,NaN,California,Binders,NaN,APAC,NaN
freq,31273,915,9994,40,108,NaN,11002,NaN,135,14,29433,35,227,NaN,NaN,11117,NaN,NaN,26518,130,30775,NaN,2001,6152,NaN,11002,NaN
mean,NaN,NaN,NaN,NaN,NaN,0.142908,NaN,1.0,NaN,NaN,NaN,NaN,NaN,28.610982,3.476545,NaN,25645.50000,246.498440,NaN,NaN,NaN,26.375818,NaN,NaN,2012.777208,NaN,31.287112
std,NaN,NaN,NaN,NaN,NaN,0.212280,NaN,0.0,NaN,NaN,NaN,NaN,NaN,174.340972,2.278766,NaN,14806.29199,487.567175,NaN,NaN,NaN,57.296810,NaN,NaN,1.098931,NaN,14.429795
min,NaN,NaN,NaN,NaN,NaN,0.000000,NaN,1.0,NaN,NaN,NaN,NaN,NaN,-6599.978000,1.000000,NaN,1.00000,0.000000,NaN,NaN,NaN,0.002000,NaN,NaN,2011.000000,NaN,1.000000
25%,NaN,NaN,NaN,NaN,NaN,0.000000,NaN,1.0,NaN,NaN,NaN,NaN,NaN,0.000000,2.000000,NaN,12823.25000,31.000000,NaN,NaN,NaN,2.610000,NaN,NaN,2012.000000,NaN,20.000000
50%,NaN,NaN,NaN,NaN,NaN,0.000000,NaN,1.0,NaN,NaN,NaN,NaN,NaN,9.240000,3.000000,NaN,25645.50000,85.000000,NaN,NaN,NaN,7.790000,NaN,NaN,2013.000000,NaN,33.000000
75%,NaN,NaN,NaN,NaN,NaN,0.200000,NaN,1.0,NaN,NaN,NaN,NaN,NaN,36.810000,5.000000,NaN,38467.75000,251.000000,NaN,NaN,NaN,24.450000,NaN,NaN,2014.000000,NaN,44.000000


In [14]:
# Date Column Inspection
df['Order.Date'] = pd.to_datetime(df['Order.Date'])
print(df['Order.Date'].dtype)

datetime64[ns]


**Summary**

* The dataset consists of 51,290 records and 27 attributes. Data profiling confirmed no missing values and no duplicate records. Date columns were converted to appropriate datetime formats to support temporal aggregation.